overlord/snapstate/handlers.go

```diff
@@ -951,6 +960,14 @@ func continueInhibitedAutoRefresh(st *state.State) {
 	st.EnsureBefore(0)
 }
 
+func abortMonitoring(st *state.State, snapName string) {
+	if cachedMonitored := st.Cached("monitored-snaps"); cachedMonitored != nil {
+		if monitoredSnaps, ok := cachedMonitored.(map[string]chan<- bool); ok {
```

Comment:
I wonder if we should have a small helper snapMonitoring(st, snapName) to capture this and use it consistently where we access monitored-snaps